nixos/modules/services/networking/networkd-dispatcher.nix

```diff
@@ -103,21 +103,26 @@ in
 
     services.networkd-dispatcher.extraArgs =
       let
-        scriptDir = pkgs.symlinkJoin {
-          name = "networkd-dispatcher-script-dir";
-          paths = lib.mapAttrsToList (
-            name: cfg:
-            (map (
-              state:
-              pkgs.writeTextFile {
-                inherit name;
-                text = cfg.script;
-                destination = "/${state}.d/${name}";
-                executable = true;
-              }
-            ) cfg.onState)
-          ) cfg.rules;
-        };
+        scriptDir =
+          pkgs.runCommand "networkd-dispatcher-script-dir" { } ''
```

Comment:
```suggestion
          pkgs.runCommand "networkd-dispatcher-script-dir" { } (''
```

Reply:
This quote is required as otherwise we end up with something like this:

```
Environment="networkd_dispatcher_args=--verbose --script-dir '/nix/store/gzbssvkb46wx9wnafwcbw5j3nv341v1g-networkd-dispatcher-script-dirmkdir -p $out/routable.d\nln -s /nix/store/4p57jf4v7pzkcs7ji2gpcjx3h2885w80-50-tailscale/bin/50-tailscale $out/routable.d/50-tailscale\n'"
```